homeassistant/components/goalzero/config_flow.py

```diff
@@ -0,0 +1,85 @@
+"""Config flow for Goal Zero Yeti integration."""
+import logging
+
+from goalzero import GoalZero, exceptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_NAME, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"host": str, "name": str})
+
+
+class GoalZeroFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Goal Zero Yeti."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        return await self.async_step_init(user_input)
+
+    async def async_step_import(self, user_input=None):
+        """Handle a flow initiated by import."""
+        return await self.async_step_init(user_input, is_import=True)
```

Comment:
Shouldn't I remove this if we are not importing anything?

Reply:
If nothing is being imported it's not needed

Reply:
Thanks. Could you tell me what's wrong with my tests and how to get 100% coverage on config_flow?

Reply:
```
# pytest --cov=homeassistant/components/goalzero/ --cov-report term-missing -- tests/components/goalzero/test_*.py
Test session starts (platform: linux, Python 3.8.1, pytest 6.0.1, pytest-sugar 0.9.3)
rootdir: /root/home-assistant, configfile: setup.cfg
plugins: requests-mock-1.8.0, asyncio-0.14.0, aiohttp-0.3.0, test-groups-1.0.3, sugar-0.9.4, forked-1.3.0, xdist-2.1.0, cov-2.10.0, timeout-1.3.4
collecting ... 
 tests/components/goalzero/test_config_flow.py ✓✓                                                                                                                    100% ██████████

----------- coverage: platform linux, python 3.8.1-final-0 -----------
Name                                               Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------
homeassistant/components/goalzero/config_flow.py      45      9    80%   30, 50-55, 84-85
homeassistant/components/goalzero/const.py            11      0   100%
--------------------------------------------------------------------------------
TOTAL                                                 56      9    84%



```

Reply:
Mock `get_state` to throw the `ConnectError` exception (use side_effect) in the test.

Remove the second try/except in `_async_try_connect` since you are already catching it above.

Reply:
Done, 100%. But python 3.7 group 2 fails. Please check it out. It's talking about magicmock can't be used in await but I'm using AsyncMock. 😒 Is this just a 3.7 compatibility issue?